extension/src/prototypes/view/ui-containers/CompassButton.tsx

```diff
@@ -0,0 +1,35 @@
+import { useAtom } from "jotai";
+import { FC, useCallback, useEffect, useState } from "react";
+
+import { useCamera } from "../../../shared/reearth/hooks";
+import { AppIconButton, CompassIcon } from "../../ui-components";
+import { autoRotateCameraAtom } from "../states/app";
+
+export const CompassButton: FC = () => {
+  const [rotationAngle, setRotationAngle] = useState(0);
+
+  const { getCameraPosition, flyTo } = useCamera();
+  const camera = getCameraPosition();
+  const radianToDegree = useCallback((rad: number) => rad * (180 / Math.PI), []);
+
+  const [initialCamera] = useState(camera);
+
+  useEffect(() => {
+    if (camera?.heading) {
+      setRotationAngle(360 - radianToDegree(camera?.heading));
+    }
+  }, [camera?.heading, radianToDegree]);
+
+  const [autoRotateCamera, setAutoRotateCameraAtom] = useAtom(autoRotateCameraAtom);
+
+  const handleClick = useCallback(() => {
+    if (initialCamera) flyTo({ ...camera, heading: 0 });
```

Comment:
Why using `initialCamera` for condition but `camera` for value here?

Reply:
@airslice sorry I forget to remove it. updated now